# Exercise 5 - Testing the hypothesis about Bias and Regression – Rocket Data

# Objective - Testing the hypothesis for rocket data

# Problem
# A rocket motor is manufactured by bonding an igniter propellant and a sustainer propellant together inside a metal housing. The shear strength of the bond between the two types of propellant is an important quality characteristic. It is suspected that shear strength is related to the age in weeks of the batch of sustainer propellant.

# (1) Fit a simple linear regression model for the rocket propellant data
# (2) Compute the estimated error variance
# (3) Compute the standard errors of estimates
# (4) Test the significance of regression coefficient and bias
# (5) Construct the 95% confidence intervals and offfer your comments

In [1]:
import pandas as pd
df=pd.read_csv("rocket_data.csv")
#df.head()

# Step 1 : Slicing of regressor and reponse columns

In [2]:
X=df[['AgeofPropellant']].values # regressor column
y=df[['ShearStrength']].values # response column

# Step 2 : Fittting the simple linear regression model

In [3]:
from sklearn.linear_model import LinearRegression
model=LinearRegression() # creating an instance of LinearRegression
model.fit(X,y)

LinearRegression()

# Step 3: Printing the estimated regression coefficient and bias

In [4]:
b0=model.intercept_[0]
b1=model.coef_[0,0]
print("Estimated bias and regression coefficients are ",b0,b1)

Estimated bias and regression coefficients are  2627.822359001296 -37.15359094490525


# Step 4: Estimated Variance

In [5]:
y_pred=model.predict(X)
SS_Res=sum((y-y_pred)*(y-y_pred))
n=len(X)

In [6]:
est_var=SS_Res/(n-2)
est_var=est_var[0]
print("Estimated Variance is ",est_var)

Estimated Variance is  9236.38100372117


In [7]:
import numpy as np
S_xx=sum((X-np.mean(X))*(X-np.mean(X)))
S_xx=S_xx[0]

# Step 5 : Computing the standard errors

In [8]:
xbar=np.mean(X)
seb0=np.sqrt(est_var*((1/n)+(xbar*xbar)/S_xx))
seb1=np.sqrt(est_var/S_xx)
print('Standard errors of bo_hat and b1_hat are', seb0,'....',seb1)

Standard errors of bo_hat and b1_hat are 44.18391179817861 .... 2.8891065489219176


# Step 6 : Test $H:\beta_1=0$ : Test statistic is $T=\frac{\hat\beta_1}{se(\hat\beta_1)}$

In [9]:
T=np.abs(b1)/seb1
print("The test statistic value is ",T)

The test statistic value is  12.85988949032332


In [10]:
from scipy.stats import t
alpha=0.05
t_tab=t.ppf(1-(alpha/2),n-2)
print("t-table value is ",t_tab)


t-table value is  2.10092204024096


# The hypothesis is rejected. That is , we conclude that the regression coefficient is significantly different from zero

# Step 7 : Construct the confidence interval for $\beta_1$

In [11]:
LCL=b1-t_tab*seb1
UCL=b1+t_tab*seb1
print("Confidnce interval for beta 1 is", "(",LCL,UCL,")",)

Confidnce interval for beta 1 is ( -43.2233785701398 -31.083803319670693 )


# Step 8 : Test $H:\beta_1=-40$  : Test statistic is $T=|\frac{\hat\beta_1-\beta_{10}}{se(\hat\beta_1)}|$

In [12]:
T=np.abs(b1-(-40))/seb1
print("The test statistic value is ",T)

The test statistic value is  0.9852212117815118


# The hypothesis is accepted. That is, the true regression coefficient is not significantly diffferent from -40. Note that acceptance will be the conclusion if we try any value which lies in the confidence interval

# Step 9 : Test $H:\beta_0=0$ : Test statistic is $T=\frac{\hat\beta_0}{se(\hat\beta_0)}$

In [13]:
T=np.abs(b0)/seb0
print("The test statistic value is ",T)

The test statistic value is  59.47464251251792


# The hypothesis is rejected. That is , we conclude that the bias coefficient is significantly different from zero

# Conclusion:

## (1) The fitted model is 2627.822359001296 - 37.15359094490525 x
## (2) Estimated error variance is 9236.38100372117
## (3) The standard errors of estimates of $\beta_0$ and $\beta_1$ is 44.18391179817861 and 2.8891065489219176
## (4) The hypothesis is accepted for regression coefficient. That is, the true regression coefficient is not significantly diffferent from -40. The hypothesis is rejected for bias coefficient. That is , we conclude that the bias coefficient is significantly different from zero
## (5) Confidence Interval of $\beta_1$ is ( -43.2233785701398 -31.083803319670693 )




